packages/frontend/apps/electron/src/main/protocol.ts

```diff
@@ -148,22 +170,33 @@ export function registerProtocol() {
       return;
     }
 
-    // session cookies are set to file:// on production
-    // if sending request to the cloud, attach the session cookie (to affine cloud server)
-    if (isNetworkResource(pathname) && sameSite) {
-      const cookie = getCookies();
-      if (cookie) {
-        const cookieString = cookie.map(c => `${c.name}=${c.value}`).join('; ');
+    (async () => {
+      // session cookies are set to file:// on production
+      // if sending request to the cloud, attach the session cookie (to affine cloud server)
+      if (
+        url.protocol === 'http:' ||
+        url.protocol === 'https:' ||
+        url.protocol === 'ws:' ||
+        url.protocol === 'wss:'
+      ) {
+        const cookies = await session.defaultSession.cookies.get({
+          url: details.url,
+        });
+
+        const cookieString = cookies
```

Comment:
Need to consider cases that the request headers already have `Cookie`/`cookie` fields.